smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -65,6 +70,173 @@ void writeMembers(TypeScriptWriter writer, Shape shape) {
         }
     }
 
+    void writeFilterSensitiveLog(TypeScriptWriter writer, String objectParam) {
+        writer.write("...$L,", objectParam);
+        for (MemberShape member : members) {
+            if (isMemberOverwriteRequired(member)) {
+                Shape memberTarget = model.expectShape(member.getTarget());
+                String memberName = getSanitizedMemberName(member);
+                String memberParam = String.format("%s.%s", objectParam, memberName);
+                writer.openBlock("...($1L.$2L && { $2L: ", "}),", objectParam, memberName, () -> {
+                    if (member.getMemberTrait(model, SensitiveTrait.class).isPresent()) {
+                        // member is Sensitive, hide the value.
+                        writer.write("SENSITIVE_STRING");
+                    } else if (memberTarget instanceof StructureShape) {
```

Comment:
How do `UnionShape` members get filtered?

Reply:
This is a PR for adding filterSensitiveLog to Structure namespaces.
The filterSensitiveLog method for Union shape would be added in a separate PR.